src/drivers/actuators/vertiq_io/vertiq_configuration_handler.cpp

```diff
@@ -0,0 +1,144 @@
+	_client_manager->AddNewClient(_pulsing_rectangular_input_parser_client);
+#endif //CONFIG_USE_PULSING_CONFIGURATION
+}
+
+void VertiqConfigurationHandler::InitClientEntryWrappers(){
+	AddNewClientEntry<float, float>(param_find("MAX_VELOCITY"), &(_prop_input_parser_client->velocity_max_));
+	AddNewClientEntry<float, float>(param_find("MAX_VOLTS"), &(_prop_input_parser_client->volts_max_));
+	AddNewClientEntry<uint8_t, int32_t>(param_find("CONTROL_MODE"), &(_prop_input_parser_client->mode_));
+	AddNewClientEntry<uint8_t, int32_t>(param_find("VERTIQ_MOTOR_DIR"), &(_prop_input_parser_client->sign_));
+	AddNewClientEntry<uint8_t, int32_t>(param_find("VERTIQ_FC_DIR"), &(_prop_input_parser_client->flip_negative_));
+
+#ifdef CONFIG_USE_IFCI_CONFIGURATION
+	AddNewClientEntry<uint8_t, int32_t>(param_find("THROTTLE_CVI"), &(_ifci_client->throttle_cvi_));
+#endif //CONFIG_USE_IFCI_CONFIGURATION
+
+#ifdef CONFIG_USE_PULSING_CONFIGURATION
+	AddNewClientEntry<uint8_t, int32_t> (param_find("PULSE_VOLT_MODE"),
+			&(_pulsing_rectangular_input_parser_client->pulsing_voltage_mode_));
+	AddNewClientEntry<uint8_t, int32_t>(param_find("X_CVI"), &(_ifci_client->x_cvi_));
+	AddNewClientEntry<uint8_t, int32_t>(param_find("Y_CVI"), &(_ifci_client->y_cvi_));
+	AddNewClientEntry<float, float>(param_find("ZERO_ANGLE"), &(_voltage_superposition_client->zero_angle_));
+	AddNewClientEntry<float, float>(param_find("VELOCITY_CUTOFF"),
+			&(_voltage_superposition_client->velocity_cutoff_));
+	AddNewClientEntry<float, float>(param_find("TORQUE_OFF_ANGLE"),
+			&(_voltage_superposition_client->propeller_torque_offset_angle_));
+	AddNewClientEntry<float, float>(param_find("PULSE_VOLT_LIM"),
+						&(_pulsing_rectangular_input_parser_client->pulsing_voltage_limit_));
+#endif //CONFIG_USE_PULSING_CONFIGURATION
+}
+
+void VertiqConfigurationHandler::UpdateClientsToNewObjId(uint8_t new_object_id)
+{
+	_object_id_now = new_object_id;
+
+	DestroyAndRecreateClient<EscPropellerInputParserClient>(_prop_input_parser_client, new_object_id);
+
+#ifdef CONFIG_USE_IFCI_CONFIGURATION
+	DestroyAndRecreateClient<IQUartFlightControllerInterfaceClient>(_ifci_client, new_object_id);
+#endif
+
+#ifdef CONFIG_USE_PULSING_CONFIGURATION
+	DestroyAndRecreateClient<VoltageSuperPositionClient>(_voltage_superposition_client, new_object_id);
+	DestroyAndRecreateClient<PulsingRectangularInputParserClient>(_pulsing_rectangular_input_parser_client, new_object_id);
+#endif
+}
+
+void VertiqConfigurationHandler::MarkConfigurationEntriesForRefresh()
+{
+	for (uint8_t i = 0; i < _added_configuration_entry_wrappers; i++) {
+		_configuration_entry_wrappers[i]->SetNeedsInit();
+	}
+}
+
+void VertiqConfigurationHandler::UpdateIquartConfigParams()
+{
+	for (uint8_t i = 0; i < _added_configuration_entry_wrappers; i++) {
+		_configuration_entry_wrappers[i]->SendGet(_serial_interface);
+		//Ensure that these get messages get out
+		_serial_interface->ProcessSerialTx();
```

Comment:
Could you ask the client manager to handle the communication for you here instead of talking to the serial interface itself? That way this thing wouldn't need the serial interface at all, right? Just needs a manager

Reply:
It makes sense to let the client manager handle this, but we still need the pointer to the serial interface since the entries need one for the gets/sets